src/ansys/dpf/core/mesh_info.py

```diff
@@ -0,0 +1,194 @@
+"""
+MeshInfo
+==========
+"""
+from ansys.dpf.core import server as server_module
+from ansys.dpf.core.generic_data_container import GenericDataContainer
+from ansys.dpf.core.scoping import Scoping
+from ansys.dpf.core.string_field import StringField
+
+
+class MeshInfo:
+    """Represents the mesh information.
+
+    This class describes the available mesh information.
+
+    Parameters
+    ----------
+     server : ansys.dpf.core.server, optional
+        Server with the channel connected to the remote or local instance.
+        The default is ``None``, in which case an attempt is made to use the
+        global server.
+
+    Examples
+    --------
+    Explore the mesh info from the model
+
+    >>> from ansys.dpf import core as dpf
+    >>> from ansys.dpf.core import examples
+    >>> fluent = examples.fluid_axial_model()
+    >>> model = dpf.Model(fluent)
+    >>> # mesh_info = model.metadata.mesh_info # printable mesh_info
+
+    """
+
+    def __init__(
+        self,
+        generic_data_container=None,
+        mesh_info=None,
+        server=None,
+    ):
+        """Initialize with a MeshInfo message"""
+        # ############################
+        # step 1: get server
+        self._server = server_module.get_or_create_server(server)
+
+        try:
+            if generic_data_container is None and mesh_info is None:
+                self._generic_data_container = GenericDataContainer()
+            elif generic_data_container is not None and mesh_info is None:
+                self._generic_data_container = generic_data_container
+            elif generic_data_container is None and MeshInfo is not None:
+                self._generic_data_container = mesh_info.generic_data_container
+        except ValueError:
+            print("Both generic data container and mesh info can't be filled")
+
+    def __call__(self):
```

Comment:
This was because of the way we implemented in a first place the mesh_info_provider. With the latest changes we made on this class, I don't see any reason to keep this one